tap_twitter_ads/__init__.py

```diff
@@ -20,10 +23,34 @@
     "consumer_secret",
 ]
 
+CREDENTIALS_FILENAME = "twitter.json"
+
+
+def get_credentials():
+    """ Get credentials """
+
+    credential_path = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), CREDENTIALS_FILENAME
+    )
+    credentials = ""
+    if os.path.exists(credential_path):
+        with open(credential_path, "r") as file:
+            twitter_credentials = file.read()
+            if twitter_credentials:
+                credentials = json.loads(twitter_credentials)
+
+    if not credentials or credentials == "":
+        # Get new credentials
+
+    return credentials
```

Comment:
While perfectly correct, using nested if-with-if like this makes it rather hard to reason about.

Consider instead using guard-clauses like below. It essentially amounts to having the happy path at the root of the function, and error checks etc. handled in ifs. This means you are essentially whittling the potential error cases away as you read further down. It is also easier for the eye to track on the vertical axis than the horizontal one.

Also, `None` is a perfectly reasonably "variable does not have a value yet" default. Although the below version does in fact not need it:

```suggestion
    if not os.path.exists(credential_path):
        return None
     
     with open(credential_path, "r") as file:
           return json.load(file)

    return None
```